src/Model/Casts/DateTime/DateTimeImmutableCast.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Mongolid\Model\Casts\DateTime;
+
+use DateTimeImmutable;
+use MongoDB\BSON\UTCDateTime;
+use Mongolid\Util\LocalDateTime;
+
+class DateTimeImmutableCast extends BaseDateTimeCast
```

Comment:
Fica a sugestão de usar a mesma nomenclatura que usamos na string da configuração.


```suggestion
class ImmutableDateTimeCast extends BaseDateTimeCast
```